istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,564 @@
+istioctl experimental post-install webhook enable --validation --validation-secret istio.webhook.galley 
+    --namespace istio-system --config-path validatingwebhookconfiguration.yaml --ca-bundle-file ./k8s-ca-cert.pem
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// --namespace is used as the namespace of the validation secret
+			opts.validationSecretNameSpace = namespace
+			if len(opts.webhookConfigPath) == 0 {
+				return fmt.Errorf("must specify a valid --config-path")
+			}
+			if len(opts.validationSecretName) == 0 {
+				return fmt.Errorf("must specify a valid --validation-secret")
+			}
+			if !opts.enableValidationWebhook {
+				return fmt.Errorf("not enabling validation webhook")
+			}
+			client, err := createInterface(kubeconfig)
+			if err != nil {
+				return fmt.Errorf("err when creating k8s client interface: %v", err)
+			}
+			// Read k8s CA certificate that will be used as the CA bundle of the webhook configuration
+			caCert, err := readCACert(client, opts.caCertPath, opts.validationSecretName, opts.validationSecretNameSpace,
+				opts.maxTimeForReadingWebhookCert)
+			if err != nil {
+				return fmt.Errorf("err when reading CA cert: %v", err)
+			}
+			fmt.Printf("Webhook CA certificate:\n%v\n", string(caCert))
+			// Apply the webhook configuration when the Galley webhook server is running.
+			err = waitForServerRunning(client, namespace, opts.validatingWebhookServiceName,
+				opts.maxTimeForCheckingWebhookServer)
+			if err != nil {
+				return fmt.Errorf("err when checking webhook server: %v", err)
+			}
+			webhookConfig, err := buildValidatingWebhookConfig(
+				caCert,
+				opts.webhookConfigPath,
+				opts.validatingWebhookConfigName,
+			)
+			if err != nil {
+				return fmt.Errorf("err when build validatingwebhookconfiguration: %v", err)
+			}
+			_, err = createValidatingWebhookConfig(client, webhookConfig)
+			if err != nil {
+				return fmt.Errorf("error when creating validatingwebhookconfiguration: %v", err)
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.enableValidationWebhook, "validation", true, "Specifies whether enabling"+
+		"the validating webhook.")
+	// Specifies the local file path to webhook CA bundle.
+	// If empty, will read CA bundle from default k8s service account.
+	flags.StringVar(&opts.caCertPath, "ca-bundle-file", "",
+		"Specifies the local file path to the webhook certificate signing CA bundle. "+
+			"If the certificate is signed by k8s CA, this parameter needs to "+
+			"be consistent with the k8s CA signing certificate, which may be customized through --cluster-signing-cert-file "+
+			"(the Kubernetes CA certificate used to sign certificates on kube-controller-manager, details in "+
+			"https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/). In the case "+
+			"that k8s signing certificate is the same as that of the k8s API server, this parameter may be configured as "+
```

Comment:
Use Kubernetes, not k8s.  Everywhere on this PR.  https://istio.io/about/contribute/style-guide/#miscellaneous

Reply:
Done.